### PR TITLE
Nic's HW4

### PR DESCRIPTION
**Pull request by Nic-Mon**

### CODE REVIEW

File: HW04_ch07_ex02.py [hunk 1]
```diff
@@ -24,12 +24,26 @@
 
 
 # Body
+def eval_loop():
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: what if my first input is `done`

File: HW04_ch08_ex05.py [hunk 2]
```diff
@@ -2,3 +2,33 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+#rotates letter c by int n
+def rotate_letter(c, n):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good job.

**Comment by danielsgriffin** [comment]
Good job.

File: D04_ex00.py [hunk 3]
```diff
@@ -11,18 +11,37 @@
...
+				print("Must be between 1 and 25")
+			elif(guess < x):
+				print("Too Low")
+			elif(guess > x):
+				print("Too High")
+		except:
+			print("Not an int!")
+		tries += 1
+	else:
+		print("Too many tries, the number was " + str(x))
```
Comment (OhRobin) [inline D04_ex00.py]: I believe this should still work with 
`print("Too many tries, the number was " + x)`

File: HW04_ch08_ex00.py [hunk 4]
```diff
@@ -20,14 +20,24 @@
 
 
 # Body
-
+def count(s, c):
```
Comment (OhRobin) [inline HW04_ch08_ex00.py]: c stands for character? Descriptive naming would be helpful

File: HW04_string2.py [hunk 5]
```diff
@@ -17,8 +17,12 @@
 # If the string length is less than 3, leave it unchanged.
 # Return the resulting string.
 def verbing(s):
-    # +++your code here+++
-    return
+    if(len(s) >= 3):
```
Comment (OhRobin) [inline HW04_string2.py]: Very concise!

File: HW04_string2.py [hunk 6]
```diff
@@ -30,8 +34,11 @@ def verbing(s):
 # So 'This dinner is not that bad!' yields:
 # This dinner is good!
 def not_bad(s):
-    # +++your code here+++
-    return
+    not_index = s.find('not')
+    bad_index = s.find('bad')
+    if(not_index < bad_index):
+        s = s.replace(s[not_index:bad_index+3],'good')
```
Comment (OhRobin) [inline HW04_string2.py]: I don't know why +3 got in here?

**Pull request closed without merging**